build - fix build environment sourcing compatibility with zsh and bash

albinsjolund opened the pull request.

lazan reviewed (changes requested):
Fix commit message with what you've done. I know it's pretty much the subject in other words, but practice makes perfect 😄 
Also, reference the issue in the commit. `fixes #85`  and NOT in the subject line. Every byte counts in the subject and it would be insane to pollute it with stuff that does not matter over time for the commit. The issue is of course interested in the relation during it's lifetime. 

Force push, of course.

albinsjolund committed:
build - fix build environment sourcing compatibility with zsh and bash

Bash and zsh uses different syntax for indirect expansion.
With "eval" we can achive the same thing in both shells with the same syntax.
(eval should work in all POSIX and most Bourne shells, but for now only tested in bash and zsh)

Fixes #85

lazan reviewed (approved):
Nice!
